format/FormatNumber.ts

```diff
@@ -430,14 +430,19 @@ function calcStyleFromColorSpec(v: number, colorSpec: ColorSpec | boolean): CSSP
 }
 
 function buildFormatConfig(v, precision, zeroPad, withCommas, omitFourDigitComma): Numbro.Format {
-    const num = Math.abs(v);
+    const num = Math.abs(v),
+        decimalCount = countDecimalPlaces(num);
 
     const config: Numbro.Format = {};
     let mantissa = undefined;
 
     if (precision % 1 === 0) {
         precision = precision < MAX_NUMERIC_PRECISION ? precision : MAX_NUMERIC_PRECISION;
-        mantissa = precision === 0 ? 0 : precision;
+        if (typeof zeroPad === 'number' && zeroPad < precision) {
+            mantissa = decimalCount < zeroPad ? zeroPad : decimalCount;
```

Comment:
We could compute `decimalCount` within this branch only. We want to ensure this method has as little overhead as possible - it can/will be called many, many, many times and we don't want to do any work that we don't need to do while formatting each number.